src/battery-check/battery-check.c

```diff
@@ -153,9 +136,9 @@ static int run(int argc, char *argv[]) {
 
         log_setup();
 
-        r = proc_cmdline_parse(parse_proc_cmdline_item, NULL, PROC_CMDLINE_STRIP_RD_PREFIX);
+        r = proc_cmdline_get_bool_full("systemd.battery-check", PROC_CMDLINE_STRIP_RD_PREFIX, /* skip_missing = */ true, &arg_doit);
         if (r < 0)
```

Comment:
Then, `if (r < 0 && r != -ENOKEY)`.

Reply:
Reconsidering it, the logic seems weird to first specify `PROC_CMDLINE_MISSING_TO_FATAL` and then ignore the error. And the change got way bigger than I originally imagined...